Sources/Core/View/SwiftUI/SnackbarView.swift

```diff
@@ -0,0 +1,248 @@
+//
+//  SnackbarView.swift
+//  SparkSnackbar
+//
+//  Created by louis.borlee on 13/09/2024.
+//  Copyright © 2024 Adevinta. All rights reserved.
+//
+
+import SwiftUI
+import Combine
+import SparkTheming
+import SparkButton
+@_spi(SI_SPI) import SparkCommon
+
+public struct SnackbarView<SnackbarButton>: View where SnackbarButton: View  {
+
+    public struct ButtonConfiguration {
+        public let intent: ButtonIntent
+        public let variant: ButtonVariant
+        public let size: ButtonSize
+        public let shape: ButtonShape
+    }
+
+    private let image: Image?
+    private let text: () -> Text
+
+    private var type: SnackbarType
+    private var variant: SnackbarVariant
+
+    @ViewBuilder private var button: () -> SnackbarButton?
+
+    @ObservedObject private var viewModel: SnackbarViewModel
+
+    @ScaledMetric private var iconSize = SnackbarConstants.iconSize
+    @ScaledMetric private var cornerRadius: CGFloat
+
+    private init(
+        theme: any Theme,
+        intent: SnackbarIntent,
+        type: SnackbarType,
+        variant: SnackbarVariant,
+        image: Image?,
+        text: @escaping () -> Text,
+        button: ((ButtonView) -> SnackbarButton)? = nil,
+        action: (() -> Void)? = nil,
+        buttonConfiguration: ((SnackbarView.ButtonConfiguration) -> SnackbarButton)? = nil
+    ) {
+        self.type = type
+        self.variant = variant
+
+        let viewModel = SnackbarViewModel(
+            theme: theme,
+            variant: variant,
+            intent: intent
+        )
+        self.viewModel = viewModel
+        self._cornerRadius = .init(wrappedValue: viewModel.theme.layout.spacing.medium)
```

Comment:
Can be managed by the VM ? (directly return the corner radius) ? I don't know :D